CMakeLists.txt

```diff
@@ -130,7 +120,19 @@ if(CMAKE_C_COMPILER_ID MATCHES "GNU|AppleClang|Clang")
 
   # These flags are just supported on some of the compilers, so we
   # check them before adding them.
-  add_compile_options_if_supported(-Wno-format-truncation -Wimplicit-fallthrough)
+  check_c_compiler_flag(-Wno-format-truncation CC_SUPPORTS_NO_FORMAT_TRUNCATION)
+  if(CC_SUPPORTS_NO_FORMAT_TRUNCATION)
+    add_compile_options(-Wno-format-truncation)
+  else()
+    message(STATUS "Compiler does not support -Wno-format-truncation")
```

Comment:
Maybe there is a way to generate such messages automatically by cmake?

Reply:
I had a function before, to make this more automatic, but for some reason this fails. It requires some more investigations, doing it this way for now.